pkg/virt-handler/virtwrap/api/converter.go

```diff
@@ -20,11 +20,12 @@ type ConverterContext struct {
 	VirtualMachine *v1.VirtualMachine
 }
 
-func Convert_v1_Disk_To_api_Disk(diskDevice *v1.Disk, disk *Disk) error {
+func Convert_v1_Disk_To_api_Disk(diskDevice *v1.Disk, disk *Disk, machine string, index int) error {
 
 	if diskDevice.Disk != nil {
 		disk.Device = "disk"
-		disk.Target.Device = diskDevice.Disk.Device
+		disk.Target.Bus = diskBusFromMachine(diskDevice.Disk.Bus, machine)
+		disk.Target.Device = diskDeviceFromBusAndIndex(diskDevice.Disk.Device, disk.Target.Bus, index)
```

Comment:
I know, doing this in converter.go was my suggestion, but I think we need to do this on the cluster level. Also that q35 should be set as default on the cluster level. I know that inter-dependencies between e.g. machine type and disk are not so nice to express in the cluster-level defaulter, but you can squeeze it probably into `SetDefault_VirtualMachine`.

Reply:
I agree with the concept, but does this default fixing belong here or is better be served by another PR?

Reply:
I think it makes sense to properly implement it right right now from the defaulting standpoint. No need for extra PRs.

Reply:
Hm. If we do this on the cluster level, then it will end up in any object after creation.

And afaik we are not doing this with anything else yet, are we?